variable/test/variable_special_values_test.cpp

```diff
@@ -0,0 +1,194 @@
+#include <gtest/gtest.h>
+#include <vector>
+
+#include "fix_typed_test_suite_warnings.h"
+#include "test_macros.h"
+
+#include "scipp/core/element/special_values.h"
+#include "scipp/variable/arithmetic.h"
+#include "scipp/variable/special_values.h"
+#include "scipp/variable/variable.h"
+
+using namespace scipp;
+using namespace scipp::core;
+using namespace scipp::variable;
+
+template <typename T> constexpr auto values_for_special_value_tests() {
+  return std::vector{static_cast<T>(0.0),
+                     static_cast<T>(-1.23),
+                     static_cast<T>(3e4),
+                     std::numeric_limits<T>::quiet_NaN(),
+                     std::numeric_limits<T>::signaling_NaN(),
+                     std::numeric_limits<T>::infinity(),
+                     -std::numeric_limits<T>::infinity()};
+}
+
+template <typename T>
+class VariableSpecialValueTest : public ::testing::Test {};
+using FloatTypes = ::testing::Types<double, float>;
+TYPED_TEST_SUITE(VariableSpecialValueTest, FloatTypes);
+
+TYPED_TEST(VariableSpecialValueTest, isnan) {
+  for (TypeParam x : values_for_special_value_tests<TypeParam>()) {
+    EXPECT_EQ(isnan(x * units::m), element::isnan(x) * units::dimensionless);
```

Comment:
On windows I think the `isnan` here is being evaluated as std::isnan rather than `scipp::variable::isnan` as I assume it is supposed to. Don's ask me why it is taking this resolution but explicitly switching to `scipp::variable::isnan` solved things for me.

Reply:
Strange! :confused: Can you commit that change or should I do it?